pygmt/datasets/map_tiles.py

```diff
@@ -0,0 +1,119 @@
+"""
+Function to load raster basemap tiles from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_map_tiles"): ["contextily"]}
+
+
+def load_map_tiles(region, source=None, lonlat=True, **kwargs):
+    """
+    Load a georeferenced raster basemap from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
+
+    Parameters
+    ----------
+    region : list
+        The bounding box of the map in the form of a list [*xmin*, *xmax*,
+        *ymin*, *ymax*]. These coordinates should be in longitude/latitude if
+        ``lonlat=True`` or Spherical Mercator (EPSG:3857) if ``lonlat=False``.
+
+    source : xyzservices.TileProvider or str
```

Comment:
`contextily` provides the `contextily.providers` class, which is a wrapper of the `xyzservice.TileProvider`. Perhaps we should only mention `contextily.providers` here so users don't have to know the `xyzservices` pacakges.

Reference: https://contextily.readthedocs.io/en/latest/providers_deepdive.html

Reply:
Hmm, I tried using `source : contextily.providers or str`, but the intersphinx doesn't work because `contextily.providers` is not documented in their API reference at https://contextily.readthedocs.io/en/latest/reference.html. Maybe just stick with `xyzservices`?

Reply:
> Maybe just stick with `xyzservices`?

OK.